perun/check/factory.py

```diff
@@ -279,3 +314,78 @@ def get_strategies_for(profile):
             method = parse_strategy(strategy['method'])
             already_applied_strategies.append(method)
             yield method
+
+
+def run_detection_for_profiles(detection_method, base_profile, targ_profile, models_strategy):
```

Comment:
It is the confusing name of the function. This function is called by detection methods, which works with all model kinds, i.e. methods **Integral Comparison** and **Local Statistics**, which are called by function **degradation_in_profiles**. 

So the sequence of calls is the following: `degradation_in_profiles` -> `integral_comparison/local_statistics` -> `run_detection_with_strategy` (**new name**) -> `run_detection_for_obtained_models` (**new name**) -> `execute_analysis` at specific detection method.

Is this all right now?

Reply:
Well, it is better, let me think a little. Rename the second one to `run_detection_for_models`. I'm also thinking that this could be private (i.e. with `_` prefix) so it is clear these are mostly meant as "helper functions" and not be directly called. However, it could lead to warnings in IDE/lint/etc.

Reply:
I think the IDE/lint and som other static checkers, will not be able to find this issue because the function is out of class. Therefore I try to add the `_` prefix.